code/game/atoms.dm

```diff
@@ -307,7 +307,9 @@ its easier to just keep the beam vertical.
 	return
 
 /atom/proc/get_contained_external_atoms()
-	. = contents
+	for(var/atom/movable/AM in contents)
```

Comment:
maybe do this fancy as anything thing here? Looks like this would be called a lot

Reply:
The typecheck is intentional here since it might have weirdness or nulls or unfinished atoms in it. Saves typechecking in loops that call the proc.